src/daemon/mod.rs

```diff
@@ -428,21 +427,46 @@ pub(super) async fn start(
                         None
                     }
                 })
+                .inspect(|i| {
```

Comment:
It'd be great to de-clunk this method which is already pretty bloated. How about moving the F3 init + config + service into `f3/`?

For readability, I'd imagine the end the end usage to be along lines of:
```rust
services.spawn_block({
  f3::service(...)
})
```

After moving it into a separate module, it'd be great to have some basic unit tests if possible.

Reply:
de-clunked the method and added unit test